lib/lexer.mll

```diff
@@ -1,301 +0,0 @@
-    with _ -> try_open_in rest_of_paths fname
-  (* TODO: Put in precise location with line number and position in the whole chain of included froms? *)
-  
-  let maybe_remove_quotes str =
-    let open Core_kernel.String in
-    if is_prefix str ~prefix:"\"" &&
-       is_suffix str ~suffix:"\""
-    then drop_suffix (drop_prefix str 1) 1
-    else str
-}
-
-(* Some auxiliary definition for variables and constants *)
-let string_literal = '"' [^'"']* '"' (* TODO: We should probably expand the alphabet *)
-let identifier = ['a'-'z' 'A'-'Z' '_'] ['a'-'z' 'A'-'Z' '0'-'9' '_']* 
-
-let integer_constant =  ['0'-'9']+
-
-let exp_literal = ['e' 'E'] ['+' '-']? integer_constant
-let real_constant1 = integer_constant? '.' ['0'-'9']* exp_literal? 
-let real_constant2 = '.' ['0'-'9']+ exp_literal?
-let real_constant3 = integer_constant exp_literal 
-let real_constant = real_constant1 | real_constant2 | real_constant3
-
-rule token = parse
-(* White space, line numers and comments *)
-    '\n'                      { lexer_logger "newline" ;
-                                incr_linenum lexbuf ; token lexbuf }
-  | [' ' '\t' '\012' '\r']    { lexer_logger "space" ;
-                                token lexbuf }
-  | "/*"                      { lexer_logger "multicomment" ;
-                                multiline_comment lexbuf ; token lexbuf }
-  | "//"                      { lexer_logger "single comment" ;
-                                singleline_comment lexbuf ; token lexbuf }
-  | "#include"
-    ( [' ' '\t' '\012'
-       '\r' '\n']+)
-    ( '"' [^ '"']* '"'
-    |  [^ ' ' '\t' '\012'
-         '\r' '\n']* 
-   as fname)                  { lexer_logger ("include " ^ fname) ;
-                                let chan, path =
-                                try_open_in !include_paths
-                                  (maybe_remove_quotes fname) in
-                                let new_lexbuf = from_channel chan in
-                                (new_lexbuf).lex_start_p
-                                  <- { pos_fname= path
-                                     ; pos_lnum= 1
-                                     ; pos_bol= 0
-                                     ; pos_cnum= 0 } ;
-                                new_lexbuf.lex_curr_p
-                                  <- new_lexbuf.lex_start_p ;
-                                if dup_exists (Str.split (Str.regexp
-                                  "\" included from \"") path) then
-                                  raise (Errors.SyntaxError
-                                    (Includes (
-                                      "Found cyclical include structure.\n",
-                                      (lexeme_end_p (Stack.top include_stack)
-                                         ))))          ;
-                                Stack.push new_lexbuf include_stack ;
-                                token new_lexbuf }
```

Comment:
Could you unindent this a little, maybe by moving the { to the next line?

Reply:
OK. Fixed this.